lib/vrl/compiler/src/expression/op.rs

```diff
@@ -122,78 +121,79 @@ impl Expression for Op {
         use ast::Opcode::*;
         use value::Kind as K;
 
-        let lhs_def = self.lhs.type_def(state);
+        let mut lhs_def = self.lhs.type_def(state);
```

Comment:
There's a `remove_null` call down below that requires it.

Reply:
Another thought about this: what do you think about cloning `lhs_def` in the relevant match arm instead to mutate it locally? Otherwise this `mut` reads to me like it would expect some side effect from most branches and would be returned at a later point, which doesn't seem to be the case.